packages/components/src/components/Dialog/service/DialogServiceProvider.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { Button } from "../../Button";
+import { Typography } from "../../Typography";
+import { Dialog } from "../index";
+import { _useDialogService } from "./dialogServiceStore";
+
+/**
+ * This component is used in EDS Provider, and
+ * provides a simpler dialog interface, similar to
+ * React Native Alert
+ */
+export const DialogServiceProvider = () => {
+    const { dialogContent, finishDialog } = _useDialogService();
+    const isDialogOpen = !!dialogContent;
+    return <Dialog isOpen={isDialogOpen} isDismissable={false}>
+        <Dialog.Header>{dialogContent?.title}</Dialog.Header>
+        <Dialog.CustomContent>
+            <Typography>{dialogContent?.message}</Typography>
+        </Dialog.CustomContent>
+        <Dialog.Actions align="right">
+            {dialogContent?.buttons.map((button) =>
+                <Button
+                    key={JSON.stringify(button)}
```

Comment:
What is the reasoning for this type of keying? I just did a console.log of one of these and got

`{"text":"Dismiss","style":"destructive"}`

Meaning that we make an implicit assumption that no two buttons will be configured the same.
I know that indexes are generally discouraged, but I think it should be fine in this case seing as this list is static.

Reply:
index seem to work